lib/chef/provider/dsc_resource.rb

```diff
@@ -65,6 +66,14 @@ def define_resource_requirements
           a.whyrun err + ["Assuming a previous resource sets the RefreshMode."]
           a.block_action!
         end
+        requirements.assert(:run) do |a|
+          a.assertion { module_usage_valid? }
+          err = ["module_name must be supplied along with module_version."]
+          a.failure_message Chef::Exceptions::DSCModuleNameMissing,
+            err
+          a.whyrun err + ["Assuming a module_name is specified."]
```

Comment:
Either the module_name is supplied or it isn't, right? There isn't anything dynamic in another resource that would have specified the module_name for the user. In that case, this line should be removed.